app/models/borrow_direct_requests.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+##
+# Wrap the BorrowDirect::RequestQuery in a class that we
+# can inject our patron barcode from (and do error handling)
+class BorrowDirectRequests
+  attr_reader :patron
+  def initialize(patron)
+    @patron = patron
+  end
+
+  def requests
+    request_client.requests('open', true).map { |request| BorrowDirectRequests::Request.new(request) }.select(&:active?)
+  rescue BorrowDirect::Error
```

Comment:
Should we report these errors to honeybadger, or are they just expected?

Reply:
Unfortunately expected.  They return an error when the user has no requests (`PUBQR004: No result for query`) and always in any non-production mode since their default server always returns an error (`PUBSC003: Invalid parameter - Invalid PartnershipId`), which I presume is because they never added us to their test system.

The BorrowDirect gem doesn't make any distinction between these and returns the top level `BorrowDirect::Error`.  The gem does have some facility for setting expected error codes (although does not include this case), but the RequestQuery class has some assumptions about what is returned from the API and doesn't handle this case well.  I would be happy to submit a PR upstream to address that, which would allow us to report the real exceptions (if folks that it would be worth doing)

Reply:
👍 ; I'm happy to punt for now.